src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -20083,6 +20083,24 @@ class AuraMunchingQueue : public BasicEvent
     AuraType _auraType;
 };
 
+class ResetToHomeOrientation : public BasicEvent
+{
+public:
+    ResetToHomeOrientation(Creature& self) : _self(self) { }
+
+    bool Execute(uint64 /*eventTime*/, uint32 /*updateTime*/) override
+    {
+            if (_self.IsInWorld() && _self.FindMap() && _self.IsAlive() && !_self.IsInCombat())
+            {
+                _self.SetFacingTo(_self.GetHomePosition().GetOrientation());
+            }
+
+        return true;
+    }
+private:
+    Creature& _self;
+};
+
```

Comment:
as this file is already huge, shall we create new classes to separate files instead?

Reply:
fine by me!